cmd/cstorvolumeclaim/controller.go

```diff
@@ -385,3 +376,192 @@ func BaseLabels(cvc *apis.CStorVolumeClaim) map[string]string {
+			fmt.Sprintf("Resize already in progress %s", cvc.Name))
+
+		return fmt.Errorf("ResizeInProgress from: %v to: %v",
+			cv.Status.Capacity, cv.Spec.Capacity)
+	}
+
+	if desiredCVCSize.Cmp(cv.Spec.Capacity) > 0 {
+		if updatedCVC, err = c.markCVCResizeInProgress(cvc); err != nil {
+			glog.Errorf("failed to mark cvc %q as resizing: %v", cvc.Name, err)
+			return err
+		}
+		cvc = updatedCVC
+		// Record an event to indicate that cvc-controller is resizing this volume.
+		c.recorder.Event(cvc, corev1.EventTypeNormal, string(apis.CStorVolumeClaimResizing),
+			fmt.Sprintf("CVCController is resizing volume %s", cvc.Name))
+
+		err = c.resizeVolume(cv, desiredCVCSize)
+		if err != nil {
+			// Record an event to indicate that resize operation is failed.
+			c.recorder.Eventf(cvc, corev1.EventTypeWarning, string(apis.CStorVolumeClaimResizeFailed), err.Error())
+		}
+		return err
+	}
+	if desiredCVCSize.Cmp(cv.Status.Capacity) == 0 {
+		// Resize volume succeeded mark it as resizing finished.
+		return c.markCVCResizeFinished(cvc)
+	}
+	return nil
+}
+
+func (c *CVCController) markCVCResizeInProgress(cvc *apis.CStorVolumeClaim) (*apis.CStorVolumeClaim, error) {
+	// Mark CVC as Resize Started
+	progressCondition := apis.CStorVolumeClaimCondition{
+		Type:               apis.CStorVolumeClaimResizing,
+		LastTransitionTime: metav1.Now(),
+	}
+	newCVC := cvc.DeepCopy()
+	newCVC.Status.Conditions = MergeResizeConditionsOfCVC(newCVC.Status.Conditions,
+		[]apis.CStorVolumeClaimCondition{progressCondition})
+	return c.PatchCVCStatus(cvc, newCVC)
+}
+
+type resizeProcessStatus struct {
+	condition apis.CStorVolumeClaimCondition
+	processed bool
+}
+
+// MergeResizeConditionsOfCVC updates cvc with desired resize conditions
+// leaving other conditions untouched.
+func MergeResizeConditionsOfCVC(oldConditions, resizeConditions []apis.CStorVolumeClaimCondition) []apis.CStorVolumeClaimCondition {
+
+	resizeConditionMap := map[apis.CStorVolumeClaimConditionType]*resizeProcessStatus{}
+
+	for _, condition := range resizeConditions {
+		resizeConditionMap[condition.Type] = &resizeProcessStatus{condition, false}
+	}
+
+	newConditions := []apis.CStorVolumeClaimCondition{}
+	for _, condition := range oldConditions {
+		// If Condition is of not resize type, we keep it.
+		if _, ok := knownResizeConditions[condition.Type]; !ok {
+			newConditions = append(newConditions, condition)
+			continue
+		}
+
+		if newCondition, ok := resizeConditionMap[condition.Type]; ok {
+			newConditions = append(newConditions, newCondition.condition)
+			newCondition.processed = true
+		}
+	}
+	// append all unprocessed conditions
+	for _, newCondition := range resizeConditionMap {
+		if !newCondition.processed {
+			newConditions = append(newConditions, newCondition.condition)
+		}
+	}
+	return newConditions
+}
+
+func (c *CVCController) markCVCResizeFinished(cvc *apis.CStorVolumeClaim) error {
+	newCVC := cvc.DeepCopy()
+	newCVC.Status.Capacity = cvc.Spec.Capacity
```

Comment:
We need to update with cv status capacity instead of cvc.spec capacity

Reply:
We are updating CVC here , there is differnet func for CV status capacity

Reply:
After checking cv status and cvc spec we are deciding so not required both will already equal